portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import java.util.Iterator;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that deletes comments from Datastore. */
+@WebServlet("/delete-data")
+public final class DeleteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
+
+    PreparedQuery results = datastore.prepare(query);
+    Iterator<Entity> commentIterator = results.asIterator();
+    
+    while(commentIterator.hasNext()) {
```

Comment:
This might also be a good candidate for streams. I know we haven't talked much about streams, but they are pretty awesome and we can talk some more. Streams also touch the topic of lambdas which we can discuss in a vc with the pod.

I've left some more details below if you're interested, but feel free to go with the for-loop till we discuss and build more comfort with streams.

Often streams are extremely efficient, easy to use, and compact. A gauge of when to use them is usually how easily you can go from the for-loop version of the code to a stream analog. If it's requiring a lot of work, streams might not be fit.

So something like this might work well:
`results.asIterable().stream().forEach(entity -> datastore.delete(entity.getKey()))`

Reply:
Deleted this file and using doDelete() method in DataServlet.java file.
Definitely would love to learn about stream and use it.